Assets/Tests/InputSystem/Plugins/MockKeyboardWithUITests.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.UI;
+using UnityEngine.TestTools;
+using UnityEngine.UI;
+
+[Description("Check https://jira.unity3d.com/browse/ISXB-1561 for more details.")]
+public class MockKeyboardWithUITests : InputTestFixture
+{
+    private Keyboard m_Keyboard;
+    private GameObject m_CanvasGo;
+    private GameObject m_EventSystemGo;
+    private GameObject m_ButtonGo;
+    private InputActionAsset m_InputActionAsset;
+
+    public class CancellationHandler : MonoBehaviour, ICancelHandler
+    {
+        public Action trigger;
+        public void OnCancel(BaseEventData eventData) => trigger();
+    }
+
+    public override void Setup()
+    {
+        base.Setup();
+
+        // Add a mock keyboard
+        m_Keyboard = InputSystem.AddDevice<Keyboard>();
+
+        // first creating the UI stuff
+        m_CanvasGo = new GameObject("canvas");
+        m_ButtonGo = new GameObject("button", typeof(Button), typeof(CancellationHandler));
+        m_ButtonGo.transform.SetParent(m_CanvasGo.transform, false);
+
+        // Now creating the Input layer
+        m_EventSystemGo = new GameObject("eventSystem", typeof(EventSystem));
+        EventSystem.current.SetSelectedGameObject(m_ButtonGo);
+
+        var inputModule = m_EventSystemGo.AddComponent<InputSystemUIInputModule>();
+
+        m_InputActionAsset = ScriptableObject.CreateInstance<InputActionAsset>();
```

Comment:
Hmm now that i'm reading it, it feels like this asset will leak, I should Destroy in in TearDown